framework/src/modules/interoperability/utils.ts

```diff
@@ -593,3 +618,347 @@ export const commonCCUExecutelogic = async (args: CommonExecutionLogicArgs) => {
+		);
+	}
+
+	for (const storeKey of outboxRootStoreKeySet) {
+		if (!chainDataStoreKeySet.has(storeKey)) {
+			throw new Error(
+				`Outbox root store key ${storeKey.toString('hex')} is missing in chain data store.`,
+			);
+		}
+	}
+
+	for (const storeKey of channelDataStoreKeySet) {
+		if (!chainDataStoreKeySet.has(storeKey)) {
+			throw new Error(
+				`Channel data store key ${storeKey.toString('hex')} is missing in chain data store.`,
+			);
+		}
+	}
+
+	for (const storeKey of chainValidatorsStoreKeySet) {
+		if (!chainDataStoreKeySet.has(storeKey)) {
+			throw new Error(
+				`Chain validators store key ${storeKey.toString('hex')} is missing in chain data store.`,
+			);
+		}
+	}
+
+	const ownChainDataStoreKeySet = new dataStructures.BufferSet();
+	for (const ownChainData of genesisStore.ownChainDataSubstore) {
+		if (ownChainDataStoreKeySet.has(ownChainData.storeKey)) {
+			throw new Error(
+				`Own chain data store key ${ownChainData.storeKey.toString('hex')} is duplicated.`,
+			);
+		}
+		ownChainDataStoreKeySet.add(ownChainData.storeKey);
+
+		await ownChainAccountStore.setWithSchema(
+			ownChainData.storeKey,
+			ownChainData.storeValue,
+			ownChainAccountSchema,
+		);
+	}
+
+	const terminatedOutboxStoreKeySet = new dataStructures.BufferSet();
+	const terminatedOutboxStore = getStore(id, STORE_PREFIX_TERMINATED_OUTBOX);
+	for (const terminatedOutbox of genesisStore.terminatedOutboxSubstore) {
+		if (terminatedOutboxStoreKeySet.has(terminatedOutbox.storeKey)) {
+			throw new Error(
+				`Terminated outbox store key ${terminatedOutbox.storeKey.toString('hex')} is duplicated.`,
+			);
+		}
+		terminatedOutboxStoreKeySet.add(terminatedOutbox.storeKey);
+
+		await terminatedOutboxStore.setWithSchema(
+			terminatedOutbox.storeKey,
+			terminatedOutbox.storeValue,
+			terminatedOutboxSchema,
+		);
+	}
+
+	const terminatedStateStoreKeySet = new dataStructures.BufferSet();
+	const terminatedStateStore = getStore(id, STORE_PREFIX_TERMINATED_STATE);
+	for (const terminatedState of genesisStore.terminatedStateSubstore) {
+		const terminatedStateStoreKey = terminatedState.storeKey;
+		if (terminatedStateStoreKeySet.has(terminatedStateStoreKey)) {
+			throw new Error(
+				`Terminated state store key ${terminatedStateStoreKey.toString('hex')} is duplicated.`,
+			);
+		}
+		terminatedStateStoreKeySet.add(terminatedStateStoreKey);
+
+		const terminatedStateStoreValue = terminatedState.storeValue;
+		if (terminatedStateStoreValue.initialized === false) {
+			if (terminatedOutboxStoreKeySet.has(terminatedStateStoreKey)) {
+				throw new Error(
+					`Uninitialized account associated with terminated state store key ${terminatedStateStoreKey.toString(
+						'hex',
+					)} cannot be present in terminated outbox store.`,
+				);
+			}
+			if (
+				!terminatedStateStoreValue.stateRoot.equals(EMPTY_BYTES) ||
+				terminatedStateStoreValue.mainchainStateRoot?.length !== 32
```

Comment:
I think we shoulduse byteLength
```suggestion
				terminatedStateStoreValue.mainchainStateRoot?.byteLength !== 32
```

Reply:
I am not sure why you think this, can you maybe elaborate what's wrong with the current logic ?

Reply:
hmm not sure what was the exact reasoning, seems like both of them are returning the number of bytes in the end. @shuse2 do you remember what was the reason we wanted to use byteLength?

Reply:
we can tackle this point in a separate issue